### PR TITLE
Add GraphQL docs

### LINKED ISSUE #3741: Engine, Playground and CLI Documentation Section
### What is the issue?

We need to have a section in the docs to explains the Dagger engine in more detail

### PR DESCRIPTION
**Pull request by marcosnils**
Fixes #3741

Signed-off-by: Marcos Lilljedahl <marcosnils@gmail.com>

### CODE REVIEW

File: docs/current/cli/465058-install.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+slug: /cli/install
```
Comment (vikram-dagger) [inline docs/current/cli/465058-install.md:2]: @marcosnils 
- are we providing binaries?
- what will be the process to install for windows/linux/macos?
- are there any specific prerequisites to install/use the binaries?

File: docs/current/cli/389936-get-started.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+slug: /cli/get-started
+---
```
Comment (vikram-dagger) [inline docs/current/cli/389936-get-started.md:3]: Can you please provide an example for this get started?

File: docs/current/cli/465058-install.md (hunk 1 above)
Comment (marcosnils) [reply docs/current/cli/465058-install.md:2]: > * are we providing binaries?

yes!


> * what will be the process to install for windows/linux/macos?

Same as the previous `dagger` CLI from the Europa docs




> * are there any specific prerequisites to install/use the binaries?

not really

File: docs/current/cli/389936-get-started.md (hunk 2 above)
Comment (marcosnils) [reply docs/current/cli/389936-get-started.md:3]: > Can you please provide an example for this get started?

Sure, I'll send some examples today about the things you can do in the CLI

File: docs/current/cli/465058-install.md (hunk 1 above)
Comment (vikram-dagger) [reply docs/current/cli/465058-install.md]: Have added these steps now, please check

File: docs/current/cli/465058-install.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+# macOS ARM:
+dagger is /opt/homebrew/bin/dagger
+# macOS Intel:
+dagger is /usr/local/bin/dagger
+```
+
+If you do not have Homebrew installed, or you want to install a specific version of `dagger`, you can run:
+
+```shell
+curl -L https://dl.dagger.io/dagger/install.sh | VERSION=0.2.232 sh
```
Comment (vikram-dagger) [inline docs/current/cli/465058-install.md:52]: What version string can I use here?

File: docs/current/cli/698277-index.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  B1[" "] -.-> B2[" "] -.-> B3[" "] -.-> B4[" "]
+end
+subgraph C["your deployment pipeline"]
+  C1[" "] -.-> C2[" "] -.-> C3[" "] -.-> C4[" "]
+end
+
+lib -..-> engine -..-> oci -..-> A1 & B1 & C1
+```
+
+1. Using the Dagger CLI, your program (typically a shell script) opens a new session to a Dagger Engine: either by connecting to an existing engine, or by provisioning one on-the-fly.
```
Comment (vikram-dagger) [inline docs/current/cli/698277-index.md:55]: I've updated this sequence slightly compared to the SDK introductions. Could you please check it once?

File: docs/current/engine/990050-index.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+slug: /engine
+---
+
+# Dagger Engine
```
Comment (vikram-dagger) [inline docs/current/engine/990050-index.md:5]: Should this be Engine or API?

File: website/sidebars.js [hunk 6]
```diff
@@ -103,7 +103,54 @@ module.exports = {
...
+      collapsed: false,
+      items: [
+        {
+          type: "doc",
+          label: "Overview",
+          id: "current/engine/graphql-api",
+        },
+        {
+          type: "doc",
+          label: "Dagger Engine",
```
Comment (vikram-dagger) [inline website/sidebars.js:119]:
```suggestion
          label: "Overview",
```

File: docs/current/cli/979595-reference.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+---
+slug: /cli/979595/reference
+---
+
+# Reference
```
Comment (vikram-dagger) [inline docs/current/cli/979595-reference.md:5]: Is this the correct set of CLI options?

```
vikram@dagger:~/public/marcosnils.dagger$ ./cloak -h
Usage:
  dagger [command]

Available Commands:
  client-gen  
  completion  Generate the autocompletion script for the specified shell
  dev         
  do          
  help        Help about any command
  project     
  version     Print dagger version

Flags:
  -h, --help             help for dagger
  -p, --project string   project config file
      --workdir string   The host workdir loaded into dagger
```

**Comment by mircubed** [comment]
@jpadams @kpenfound can you add some examples here?

**Comment by vikram-dagger** [comment]
Closing in favour of https://github.com/dagger/dagger/pull/3917

**Pull request closed without merging**